src/pages/Transfer/CrossChainTransferForm/CrossChainTransferForm.tsx

```diff
@@ -35,125 +41,253 @@ import {
+    const adapter = data.bridge.findAdapter(formData[CROSS_CHAIN_TRANSFER_FROM_FIELD] as ChainName);
+    const apiPromise = data.provider.getApiPromise(formData[CROSS_CHAIN_TRANSFER_FROM_FIELD] as string);
 
-  const handleSubmit = () => {
-    console.log('submit');
-  };
+    apiPromise.setSigner(signer);
+    adapter.setApi(apiPromise);
 
-  const governanceBalance = getBalance(GOVERNANCE_TOKEN.ticker)?.free || newMonetaryAmount(0, GOVERNANCE_TOKEN);
-  const balance = getBalance(GOVERNANCE_TOKEN.ticker)?.transferable;
+    const transferAmount = newMonetaryAmount(
+      form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] || 0,
+      getCurrencyFromTicker(currentToken.value),
+      true
+    );
 
-  const schema: CrossChainTransferValidationParams = {
-    [CROSS_CHAIN_TRANSFER_AMOUNT_FIELD]: {
-      governanceBalance,
-      minAmount: newMonetaryAmount(1, GOVERNANCE_TOKEN),
-      maxAmount: balance || newMonetaryAmount(0, GOVERNANCE_TOKEN),
-      transactionFee: TRANSACTION_FEE_AMOUNT
-    }
+    const transferAmountString = transferAmount.toString(true);
+    const transferAmountDecimals = transferAmount.currency.decimals;
+
+    const tx = adapter.createTx({
+      amount: FixedPointNumber.fromInner(transferAmountString, transferAmountDecimals),
+      to: formData[CROSS_CHAIN_TRANSFER_TO_FIELD],
+      token: formData[CROSS_CHAIN_TRANSFER_TOKEN_FIELD],
+      address: formData[CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD]
+    } as CrossChainTransferParams);
+
+    await submitExtrinsic({ extrinsic: tx });
+  };
+
+  const handleSubmit = (formData: CrossChainTransferFormData) => {
+    xcmTransferMutation.mutate(formData);
   };
 
   const form = useForm<CrossChainTransferFormData>({
     initialValues: {
       [CROSS_CHAIN_TRANSFER_AMOUNT_FIELD]: '',
-      [CROSS_CHAIN_TRANSFER_FROM_FIELD]: 'kintsugi',
-      [CROSS_CHAIN_TRANSFER_TO_FIELD]: 'kusama',
-      [CROSS_CHAIN_TRANSFER_TOKEN_FIELD]: 'KSM',
+      [CROSS_CHAIN_TRANSFER_TOKEN_FIELD]: '',
       [CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD]: accountId?.toString() || ''
     },
     onSubmit: handleSubmit,
-    validationSchema: crossChainTransferSchema(schema, t)
+    validationSchema: crossChainTransferSchema(schema, t),
+    validateOnChange: false
   });
 
-  useEffect(() => {
-    if (!accountId) {
-      return form.resetForm();
+  const xcmTransferMutation = useMutation<void, Error, CrossChainTransferFormData>(mutateXcmTransfer, {
+    onSuccess: () => {
+      toast.success('Transfer successful');
+    },
+    onError: (err) => {
+      toast.error(err.message);
     }
+  });
+
+  const handleOriginatingChainChange = (chain: ChainName, name: string) => {
+    form.setFieldValue(name, chain);
+
+    const destinationChains = getDestinationChains(chain);
+
+    setDestinationChains(destinationChains);
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_FIELD, destinationChains[0].id);
```

Comment:
Instead of two individual setFieldValue, you can just do the following:

`form.setValues([name]: chain, [CROSS_CHAIN_TRANSFER_TO_FIELD]: destinationChains[0].id})`

This will only change the specified fields